### ISSUE TITLE
Directory-based analysis

### ISSUE THREAD

**Issue by gaborantal** [Persper/javascript-call-graph#13]
We can analyze several files at the same time, but only if we add them one by one as a command line argument, e.g. `node src/main.js --cg file1.js file2.js file3.js lib/dependency1.js lib/dependency2.js`. For example, Windows command prompt limits the length of the parameters (discussed [here](https://support.microsoft.com/en-us/help/830473/command-prompt-cmd-exe-command-line-string-limitation)). Furthermore, it is a really uncomfortable way to set the input files.

We should detect whether the passed argument (one of them) is a directory and if so, collect all the files in the given directory.

**Comment by UltimateBeaver** [comment]
Sounds good, added feature label, wanna get assigned to this?

**Comment by gaborantal** [comment]
Sure!

**Comment by zizo007** [comment]
Is it necessary to have an additional command line argument? A more user-friendly way would be if the system checks whether the given parameter is a file or a directory. Multiple files/directories could be passed for the tool.
What's your opinion about this?

**Comment by UltimateBeaver** [comment]
Good suggestion, I'd prefer that it automatically check whether a given param is a file or a directory. For directory parameter, is the plan to recursively collect all js files?

**Comment by zizo007** [comment]
Yes, absolutely. You only have to look for files with .js extension. Later we could add a filtering mechanism as well if somebody wants to exclude some files from a directory for example.

**Comment by gaborantal** [comment]
@zizo007 Filtering mechanism could introduce some inconsistency if we have mixed inputs. For example, one may filter out `idontcare.js` from the directories, but then add it as a file input. How the filter should work? Filter out the file we added manually or not? As we added as a file, we want to analyze that file too, but we wrote a filter that intended to filter out the file. 

However, that may be a future problem.

I have an other suggestion as well, along with this change we should also add an `--input-file` argument, which contains all the files we want to analyze, as maybe there are cases when writing a matching filter is too hard. How about that?
